lang/cs/Org.Apache.REEF.Common/Evaluator/DriverInformation.cs

```diff
@@ -89,48 +90,73 @@ public string NameServerId
+                            {
+                                var refreshKeyValue = refreshParam.Split('=').Select(kv => kv.Trim()).ToArray();
+                                if (refreshKeyValue.Length == 2 && refreshKeyValue[0].Equals("url", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    queryUri = new Uri(refreshKeyValue[1]);
+                                    break;
+                                }
+                            }
+                        }
+
+                        // We have received a redirect URI, look there instead.
+                        if (queryUri != null)
+                        {
+                            LOGGER.Log(Level.Verbose, "Received redirect URI:[" + queryUri + "], redirecting...");
+                            Thread.Sleep(500);
+                            continue;
+                        }
+
+                        Stream stream = webResponse.GetResponseStream();
+                        if (stream == null || !stream.CanRead)
+                        {
+                            return null;
+                        }
+
+                        using (StreamReader streamReader = new StreamReader(stream, Encoding.UTF8))
+                        {
+                            var driverInformation = streamReader.ReadToEnd();
+                            LOGGER.Log(Level.Verbose, "Http response line: " + driverInformation);
+                            try
+                            {
+                                var info = AvroJsonSerializer<AvroDriverInfo>.FromString(driverInformation);
+                                return info != null ? new DriverInformation(info.remoteId, info.startTime, info.services) : null;
+                            }
+                            catch (Exception e)
+                            {
+                                Utilities.Diagnostics.Exceptions.CaughtAndThrow(e, Level.Error, string.Format(CultureInfo.InvariantCulture, "Cannot read content."), LOGGER);
+                                return null;
+                            }
+                        }
                     }
                 }
-            }
-            catch (WebException)
-            {
-                LOGGER.Log(Level.Warning, string.Format(CultureInfo.InvariantCulture, "In RECOVERY mode, cannot connect to [{0}] for driver information, will try again later.", queryUri));
-                return null;
-            }
-            catch (Exception e)
-            {
-                Org.Apache.REEF.Utilities.Diagnostics.Exceptions.CaughtAndThrow(e, Level.Error, string.Format(CultureInfo.InvariantCulture, "Cannot read content from {0}.", queryUri), LOGGER);
+                catch (WebException)
+                {
+                    LOGGER.Log(Level.Warning, string.Format(CultureInfo.InvariantCulture, "In RECOVERY mode, cannot connect to [{0}] for driver information, will try again later.", queryUri));
```

Comment:
Logger accepts format strings and their parameters directly. No need to call `string.Format` here.